subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -5641,6 +5643,7 @@ int lwm2m_socket_start(struct lwm2m_ctx *client_ctx)
 			return -errno;
 		}
 
+#if defined(CONFIG_MBEDTLS_SERVER_NAME_INDICATION)
 		if (client_ctx->desthostname != NULL) {
```

Comment:
We could then check it's value here:
```suggestion
		if (client_ctx->hostname_verify && client_ctx->desthostname != NULL) {
```